tests/nnstreamer_filter_tvm/unittest_filter_tvm.cc

```diff
@@ -0,0 +1,539 @@
+  const gchar *model_files[] = {
+    model_file,
+    NULL,
+  };
+  GstTensorFilterProperties prop = {
+    .fwname = "tvm",
+    .fw_opened = 0,
+    .model_files = model_files,
+    .num_models = 1,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("tvm");
+  EXPECT_NE (sp, nullptr);
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+  EXPECT_NE (data, nullptr);
+
+  output.size = input.size = sizeof (float) * 1;
+
+  input.data = g_malloc (input.size);
+  output.data = g_malloc (output.size);
+  ((float *)input.data)[0] = 10.0;
+
+  /* unsucessful invoke */
+  ret = sp->invoke (NULL, NULL, NULL, &input, &output);
+  EXPECT_NE (ret, 0);
+
+  g_free (model_file);
+  g_free (input.data);
+  g_free (output.data);
+  sp->close (&prop, &data);
+}
+
+/**
+ * @brief Positive case with invoke for tvm model
+ */
+TEST (nnstreamerFilterTvm, invoke00)
+{
+  int ret;
+  void *data = NULL;
+  GstTensorMemory input, output;
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+
+  gchar *model_file = g_build_filename (
+      root_path, "tests", "test_models", "models", "tvm_add_one.so_", NULL);
+  ASSERT_TRUE (g_file_test (model_file, G_FILE_TEST_EXISTS));
+
+  const gchar *model_files[] = {
+    model_file,
+    NULL,
+  };
+  GstTensorFilterProperties prop = {
+    .fwname = "tvm",
+    .fw_opened = 0,
+    .model_files = model_files,
+    .num_models = 1,
+  };
+
+  output.size = input.size = sizeof (float) * 3 * 640 * 480 * 1;
+  input.data = g_malloc (input.size);
+  output.data = g_malloc (output.size);
+  memset (input.data, 0, input.size);
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("tvm");
+  EXPECT_NE (sp, nullptr);
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+  EXPECT_NE (data, (void *)NULL);
+  ret = sp->invoke (NULL, NULL, data, &input, &output);
+
+  EXPECT_EQ (ret, 0);
+  EXPECT_EQ (static_cast<float *> (output.data)[0], (float)(1.0));
+
+  g_free (model_file);
+  g_free (input.data);
+  g_free (output.data);
+  sp->close (&prop, &data);
+}
+
+/**
+ * @brief Positive case to launch gst pipeline
+ */
+TEST (nnstreamerFilterTvm, launch00)
+{
+  gchar *pipeline;
+  GstElement *gstpipe;
+  GError *err = NULL;
+  int status = 0;
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+
+  gchar *model_file = g_build_filename (
+      root_path, "tests", "test_models", "models", "tvm_add_one.so_", NULL);
+
+  /* create a nnstreamer pipeline */
+  pipeline = g_strdup_printf ("videotestsrc num-buffers=1 ! videoconvert ! videoscale ! video/x-raw,format=RGB,width=640,height=480 ! tensor_converter ! tensor_transform mode=arithmetic option=typecast:float32 ! tensor_filter framework=tvm model=\"%s\" ! fakesink",
+      model_file);
+
+  gstpipe = gst_parse_launch (pipeline, &err);
+
+  if (gstpipe) {
+    status = 0;
+
+    EXPECT_NE (gst_element_set_state (gstpipe, GST_STATE_PLAYING), GST_STATE_CHANGE_FAILURE);
```

Comment:
No need to sleep here?

Reply:
Wait for status change (there is macro or function in tests directory as a utility!) rather than sleep.

Reply:
Yes. fixed with util method